app/frontend/src/features/profile/view/leaveReference/LeaveReferencePage.tsx

```diff
@@ -67,40 +83,12 @@ export default function LeaveReferencePage() {
               </div>
             </>
           </div>
-        );
-      }
-    } else if (hostRequest) {
-      const availableReference = availableRefrences.availableWriteReferencesList.find(
-        ({ hostRequestId }) => hostRequestId === +hostRequest
-      );
-      if (availableReference) {
-        if (
-          (referenceType === "surfed" &&
-            availableReference.referenceType ===
-              ReferenceType.REFERENCE_TYPE_SURFED) ||
-          (referenceType === "hosted" &&
-            availableReference.referenceType ===
-              ReferenceType.REFERENCE_TYPE_HOSTED)
-        ) {
-          return (
-            <div className={classes.root}>
-              {error && <Alert severity="error">{error}</Alert>}
-              <>
-                <Hidden smDown>
-                  <UserOverview user={user} />
-                </Hidden>
-                <div className={classes.form}>
-                  <ReferenceForm user={user} />
-                </div>
-              </>
-            </div>
-          );
-        }
-      }
-    }
-  } else {
-    return <Alert severity="error">{REFERENCE_TYPE_NOT_AVAILABLE}</Alert>;
-  }
-
-  return <CircularProgress />;
+        ) : (
+          <Alert severity="error">{REFERENCE_TYPE_NOT_AVAILABLE}</Alert>
+        )
+      ) : (
+        <Alert severity="error">{USER_REFERENCES_UNAVAILABLE}</Alert>
```

Comment:
This is actually just `null` or use `{(availableRefrences && user) && (` instead of `{availableRefrences && user ? (`.

You already display the error at the top, and this line will be rendered while the queries are loading (`user` and `availableReferences` are undefined while they are loading).